tests/rest/client/v1/utils.py

```diff
@@ -550,6 +550,20 @@ def initiate_sso_login(
             "/_matrix/client/r0/login/sso/redirect?" + urllib.parse.urlencode(params),
         )
 
+        # that should 302 us to the public base url
+        assert channel.code == 302
+        location = channel.headers.getRawHeaders("Location")[0]
+        parts = urllib.parse.urlsplit(location)
+        channel = make_request(
+            self.hs.get_reactor(),
+            self.site,
+            "GET",
+            urllib.parse.urlunsplit(("", "") + parts[2:]),
+            custom_headers=[
+                ("Host", parts[1]),
+            ],
+        )
+
```

Comment:
this was the easiest way of figuring out what the `Host` header should be set to.